### ISSUE TITLE
364K (uncompressed) of mostly unused icons?

### ISSUE THREAD

**Issue by sdhull** [evoactivity/ember-svg-jar#154]
#### Describe your question
I was under the impression that at build-time, `{{svg-jar "plus-icon"}}` would be replaced by `<svg..>...</svg>`. Is this not the case? Should I not be adding icons to my project until they're being used?

![image](https://user-images.githubusercontent.com/50039/77218488-f561b680-6ae8-11ea-8b50-60606ee67c52.png)
![image](https://user-images.githubusercontent.com/50039/77221422-e25ddf00-6b06-11ea-8932-2471cd6b1267.png)

#### Search tags, topics
package size, compiled size

**Comment by ivanvotti** [comment]
It will include all `.svg` files that it can find in `sourceDirs` (`public` dir by default ).
So I would suggest to add only the icons you need.

I have a plan to make it possible to tell SVGJar what icons you need, so you can add all your icon sets to the SVGJar viewer without adding them to the `vendor.js` file.

**Comment by sdhull** [comment]
Yeah that would be really cool. Like a config option `onlyInclude: ['add.svg', /somepath\/.*svg/]` and if that is empty or undefined then it includes everything by default? Or would you scan for usages of `{{svg-jar` and only include icons referenced automagically?

After looking through the options again, it looks like I could at least keep all my icons in my repo and then specify `sourceDir` and only copy or move icons over as needed so not toooo bad but I wish I could have them all in the viewer but only include icons in the build that are referenced (or failing that, whitelist them as needed for inclusion in the build).

**Comment by wilkosz** [comment]
> It will include all `.svg` files that it can find in `sourceDirs` (`public` dir by default ).
> So I would suggest to add only the icons you need.
> 
> I have a plan to make it possible to tell SVGJar what icons you need, so you can add all your icon sets to the SVGJar viewer without adding them to the `vendor.js` file.

Has there been any progress with this?

**Comment by xud6** [comment]
Just hit this today.

I think every svg files is included in the final build really is an unexpected behavior nowadays, probably should say something about it in the readme

**Comment by betocantu93** [comment]
I think this could be solved with the new `hbs` strategy #186 , they are converted to template-only components at build time and downloaded when invoked in runtime, they pretty much kinda work like a normal image tag... with the benefits of splattributes and modifiers!
